client/components/widget/query/query-result-data-service.js

```diff
@@ -198,11 +199,17 @@ QueryResultDataService.prototype.fetchResults = function(datasource) {
         this.errorService_.addError(ErrorTypes.DANGER, response.data.error);
         deferred.reject(response.data);
       } else {
-        if (this.explorerService_.model.logStatistics) {
-          let rows = response.data.totalRows;
-          let size = response.data.totalBytesProcessed;
-          let speed = response.data.elapsedTime;
+        let rows = response.data.totalRows;
+        let size = response.data.totalBytesProcessed;
+        let time = response.data.elapsedTime;
+        let job = response.data.jobReference.jobId;
+
+        widget.state().datasource.job_id = job;
+        widget.state().datasource.row_count = rows;
+        widget.state().datasource.query_size = size;
+        widget.state().datasource.query_time = time;
 
+        if (this.explorerService_.model.logStatistics) {
           this.errorService_.addError(
               ErrorTypes.INFO,
               'Returned ' + this.filter_('number')(rows, 0) + ' records, ' +
```

Comment:
consider assigning `this.filter_('number')` to a "num" or "fmt" temp var to make this more readable.